viper.go

```diff
@@ -849,44 +851,45 @@ func (v *Viper) Get(key string) interface{} {
 		return nil
 	}
 
-	if v.typeByDefValue {
-		// TODO(bep) this branch isn't covered by a single test.
-		valType := val
+	valType := val
 
-		v.lock.RLock()
-		path := strings.Split(lcaseKey, v.keyDelim)
-		defVal := v.searchMap(v.defaults, path)
-		if defVal != nil {
-			valType = defVal
-		}
-		v.lock.RUnlock()
-
-		switch valType.(type) {
-		case bool:
-			val = cast.ToBool(val)
-		case string:
-			return cast.ToString(val)
-		case int32, int16, int8, int:
-			return cast.ToInt(val)
-		case uint:
-			return cast.ToUint(val)
-		case uint32:
-			return cast.ToUint32(val)
-		case uint64:
-			return cast.ToUint64(val)
-		case int64:
-			return cast.ToInt64(val)
-		case float64, float32:
-			return cast.ToFloat64(val)
-		case time.Time:
-			return cast.ToTime(val)
-		case time.Duration:
-			return cast.ToDuration(val)
-		case []string:
-			return cast.ToStringSlice(val)
-		case []int:
-			return cast.ToIntSlice(val)
-		}
+	v.lock.RLock()
+	path := strings.Split(lcaseKey, v.keyDelim)
+	valT := v.searchMap(v.types, path)
+	if valT == nil && v.typeByDefValue {
+		valT = v.searchMap(v.defaults, path)
+	}
+	v.lock.RUnlock()
+	valType = valT
+
+	switch valType.(type) {
+	case bool:
+		val = cast.ToBool(val)
+	case string:
+		return cast.ToString(val)
+	case int32, int16, int8, int:
+		return cast.ToInt(val)
+	case uint:
+		return cast.ToUint(val)
+	case uint32:
+		return cast.ToUint32(val)
+	case uint64:
+		return cast.ToUint64(val)
+	case int64:
+		return cast.ToInt64(val)
+	case float64, float32:
+		return cast.ToFloat64(val)
+	case time.Time:
+		return cast.ToTime(val)
+	case time.Duration:
+		return cast.ToDuration(val)
+	case []string:
+		return cast.ToStringSlice(val)
+	case []int:
+		return cast.ToIntSlice(val)
+	case nil:
+		// this means neither the type nor the default is taken as a type
+		// to allow also nil as a type v.searchMap would have to return a found flag
```

Comment:
This is never the case if the type "guessing" algorithm is applied correctly (see my previous comment).

```suggestion
```